### PR TITLE
Unify backend rpc addQueryParam methods

### PR DESCRIPTION
**Pull request by fm3**
Unifies the adding of query parameters to RPC objects.

Supported types can now all be passed to the same overloaded method addQueryParam. This way, callers don’t need to take care if they need addQueryString (with arrow) or addQueryStringOptional (with comma) and for supported, safe types, the caller doesn’t have to take care of the string conversion.

### Steps to test:
- CI should be enough

------
- [x] Considered [common edge cases](../blob/master/.github/common_edge_cases.md)
- [x] Needs datastore update after deployment

### CODE REVIEW

File: webknossos-datastore/app/com/scalableminds/webknossos/datastore/rpc/RPCRequest.scala [hunk 1]
```diff
@@ -25,16 +27,54 @@ class RPCRequest(val id: Int, val url: String, wsClient: WSClient)(implicit ec:
...
+    addQueryParam(key, value.toString)
+
+  def addQueryParam(key: String, value: ObjectId): RPCRequest =
+    addQueryParam(key, value.toString)
+
+  def addQueryParam(key: String, value: Boolean): RPCRequest =
+    addQueryParam(key, value.toString)
+
+  def addQueryParam(key: String, valueOptional: Option[String]): RPCRequest =
+    valueOptional.map(addQueryParam(key, _)).getOrElse(this)
```
Comment (MichaelBuessemeyer) [inline webknossos-datastore/app/com/scalableminds/webknossos/datastore/rpc/RPCRequest.scala:46]: could be a little short using fold I think. `valueOptional.fold(this)(addQueryParam(key, _))`. But as we rarely use this, this might be harder to read / understand

File: webknossos-datastore/app/com/scalableminds/webknossos/datastore/rpc/RPCRequest.scala [hunk 2]
```diff
@@ -25,16 +27,54 @@ class RPCRequest(val id: Int, val url: String, wsClient: WSClient)(implicit ec:
...
+
+  // ClassTags added to work around type erasure (otherwiese, all Option[x] variants would be indistinguishable)
```
Comment (MichaelBuessemeyer) [inline webknossos-datastore/app/com/scalableminds/webknossos/datastore/rpc/RPCRequest.scala]: typo at otherwiese

File: webknossos-datastore/app/com/scalableminds/webknossos/datastore/rpc/RPCRequest.scala [hunk 3]
```diff
@@ -25,16 +27,54 @@ class RPCRequest(val id: Int, val url: String, wsClient: WSClient)(implicit ec:
...
+  // Compare https://stackoverflow.com/a/3309490
```
Comment (MichaelBuessemeyer) [inline webknossos-datastore/app/com/scalableminds/webknossos/datastore/rpc/RPCRequest.scala:49]: Oh btw. Scala 3 would make this easier by allowing or type definition via ` A | B | C`. Just wanted to add this info.
